Source/Scene/Billboard.js

```diff
@@ -1142,7 +1142,37 @@ Billboard.prototype._loadImage = function () {
   const imageSubRegion = this._imageSubRegion;
   let imageIndexPromise;
 
+  const that = this;
+  function completeImageLoad(index) {
+    if (
+      that._imageId !== imageId ||
+      that._image !== image ||
+      !BoundingRectangle.equals(that._imageSubRegion, imageSubRegion)
+    ) {
+      // another load occurred before this one finished, ignore the index
+      return;
```

Comment:
I am trying to understand why we need to exhaustively check all three conditions here: is it because they can change independently?

Also, just double checking: we need this check because this function is not module scoped and can be triggered by an image?

Reply:
Honestly I'm not entirely sure. As far as I can see, `imageId` and `image` cannot change independently. But this just moves the logic removed below to a function so we can finish early if there is no promise to resolve, so I prefer not to modify the existing logic.